cpp/tests/column/column_view_shallow_test.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <memory>
+#include <type_traits>
+
+template <typename T>
+std::unique_ptr<cudf::column> example_column()
+{
+  // fixed_width, dict, string, list, struct
+  if constexpr (cudf::is_fixed_width<T>()) {
```

Comment:
When there are multiple branches that share essentially no code I think I'd prefer SFINAE overloads to `if constexpr` for readability.

Reply:
IMO, sfinae messes up the formatting, verbose and repeating the same code. `if constexpr` allows the code to be at a single place.

```

// fixed_width, dict, string, list, struct
template <typename T, std::enable_if_t<cudf::is_fixed_width<T>()>* = nullptr>
std::unique_ptr<cudf::column> example_column()
{
  auto begin = thrust::make_counting_iterator(1);
  auto end   = thrust::make_counting_iterator(16);
  return cudf::test::fixed_width_column_wrapper<T>(begin, end).release();
}

template <typename T, std::enable_if_t<cudf::is_dictionary<T>()>* = nullptr>
std::unique_ptr<cudf::column> example_column()
{
  return cudf::test::dictionary_column_wrapper<std::string>(
           {"fff", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "", ""}, {1, 1, 1, 1, 1, 1, 1, 1, 0})
    .release();
}

template <typename T,
          std::enable_if_t<std::is_same_v<T, std::string> or
                           std::is_same_v<T, cudf::string_view>>* = nullptr>
std::unique_ptr<cudf::column> example_column()

{
  return cudf::test::strings_column_wrapper(
           {"fff", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "", ""})
    .release();
}

template <typename T, std::enable_if_t<std::is_same_v<T, cudf::list_view>>* = nullptr>
std::unique_ptr<cudf::column> example_column()
{
  return cudf::test::lists_column_wrapper<int>({{1, 2, 3}, {4, 5}, {}, {6, 7, 8}}).release();
}

template <typename T, std::enable_if_t<std::is_same_v<T, cudf::struct_view>>* = nullptr>
std::unique_ptr<cudf::column> example_column()
{
  auto begin    = thrust::make_counting_iterator(1);
  auto end      = thrust::make_counting_iterator(16);
  auto member_0 = cudf::test::fixed_width_column_wrapper<int32_t>(begin, end);
  auto member_1 = cudf::test::fixed_width_column_wrapper<int32_t>(begin + 10, end + 10);
  return cudf::test::structs_column_wrapper({member_0, member_1}).release();
}
```

Reply:
SFINAE wins! 🥇 
updated. 👍